zerver/views/users.py

```diff
@@ -353,7 +356,7 @@ def avatar_by_id(
         if not realm.allow_web_public_streams_access():
             raise MissingAuthenticationError
 
-        if settings.RATE_LIMITING:
+        if should_rate_limit(request):
```

Comment:
Are there other places we should be making this change? This probably should be its own commit with a proper commit message explaining why it is correct in any case.